flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -667,22 +668,35 @@ public boolean containsKey(String key) {
     public boolean contains(ConfigOption<?> configOption) {
         final boolean canBePrefixMap = canBePrefixMap(configOption);
 
-        // first try the current key
         synchronized (this.confData) {
-            if (containsInternal(configOption.key(), canBePrefixMap)) {
-                return true;
-            } else if (configOption.hasFallbackKeys()) {
-                // try the fallback keys
-                for (FallbackKey fallbackKey : configOption.fallbackKeys()) {
-                    if (containsInternal(fallbackKey.getKey(), canBePrefixMap)) {
-                        loggingFallback(fallbackKey, configOption);
-                        return true;
-                    }
+            final Function<String, Optional<Boolean>> applier =
```

Comment:
Extracting that to method lambda would definitively improve readability. However, if you get rid of `Optional` it also might become a oneliner and then I wouldn't extract a method.

Reply:
But then we are back at the `contains` and `containsInternal` for every method as before.

Reply:
You are right.